runner/src/main.rs

```diff
@@ -3,22 +3,33 @@
 
 #![deny(clippy::all, clippy::pedantic)]
 
-use std::env;
+use std::path::PathBuf;
+
+use clap::{arg, value_parser, Command};
+// use std::env;
 
 fn main() -> Result<(), String> {
-    let args: Vec<String> = env::args().collect();
-    if args.len() == 1 {
-        println!("OVERVIEW: Virtual execution environment for QIR programs");
-        println!();
-        println!("USAGE: qir-runner <ll/bc filename> [entry point function]");
-        Ok(())
-    } else {
-        let path = &args[1];
-        let name = if args.len() > 2 {
-            Some(args[2].as_str())
-        } else {
-            None
-        };
-        qir_runner::run_file(path, name)
+    let cmd = Command::new("qir-runner").args(&[
+        arg!(-f --file <PATH> "(Required) Path to the QIR file to run")
+            .value_parser(value_parser!(PathBuf))
+            .required(true),
+        arg!(-e --entrypoint <NAME> "Name of the entry point function to execute"),
+        arg!(-s --shots <NUM> "The number of times to repeat the execution of the chosen entry point in the program")
+            .value_parser(value_parser!(u32))
+            .default_value("1")]);
+
+    let matches = cmd.try_get_matches().map_err(|e| e.to_string());
+    match matches {
+        Err(e) => {
+            print!("{}", e);
```

Comment:
Should this be `eprintln`?

Reply:
That's a good question. Probably? This is also how help is displayed, which feels a little odd going to stderr, but not too bad.